backend/src/rss.rs

```diff
@@ -0,0 +1,204 @@
+}
+
+/// Generates the single video items of a series in Tobira for inclusion in an RSS feed.
+async fn video_items(
+    doc: &mut ogrim::Document,
+    db: &Client,
+    series_oc_id: &str,
+    series_title: &str,
+    rss_link: &str,
+    tobira_url: &HttpHost,
+) -> Result<(), Error> {
+    let selection = Event::select();
+    let query = format!("select {selection} from events where part_of = $1");
+    let rows = db.query_raw(&query, dbargs![&series_oc_id]).await?;
+
+    rows.try_for_each(|row| {
+        let event = Event::from_row_start(&row);
+        
+        let mut buf = [0; 11];
+        let tobira_event_id = event.id.to_base64(&mut buf);
+        let event_link = format!("{tobira_url}/!v/{tobira_event_id}");
+        let thumbnail = &event.thumbnail_url.unwrap_or_default();
+        let tracks = preferred_tracks(event.tracks);
+        let enclosure_track = tracks.1.unwrap();
+        let enclosure_url = &enclosure_track.uri;
+        let mimetype = &enclosure_track.mimetype.unwrap();
+
+        xml!(doc,
+            <item>
+                <title>{event.title}</title>
+                <link>{event_link}</link>
+                <description>{event.description.unwrap_or_default()}</description>
+                <dc:creator>{event.creators.join(", ")}</dc:creator>
+                <pubDate>{event.created.to_rfc2822()}</pubDate>
+                <guid>{event_link}</guid>
+                <media:thumbnail url={thumbnail} />
+                <itunes:image href={thumbnail} />
+                <enclosure
+                    url={enclosure_url}
+                    type={mimetype}
+                    length="0"
+                />
+                <source url={rss_link}>{series_title}</source>
+                <media:group>
+                    {|doc| for track in tracks.0 {
+                        xml!(doc,
+                            <media:content
+                                url={track.uri}
+                                type={track.mimetype.unwrap()}
+                                medium={"video"}
+                                height={track.resolution.unwrap()[1]}
+                                width={track.resolution.unwrap()[0]}
+                            />
+                        )}
+                    }
+                </media:group>
+
+            </item>
+        );
+
+        future::ready(Ok(()))
+    }).await?;
+
+    Ok(())
+}
+
+
+
+/// This returns a track that:
+/// a) is a `presentation` track.
+/// Defaults to any track meeting the b) criteria if there is no `presentation` track.
+/// b) has a resolution that is closest to full hd.
+fn preferred_tracks(tracks: Vec<EventTrack>) -> (Vec<EventTrack>, Option<EventTrack>) {
+    let target_resolution = tracks.first().map_or([1920, 1080], |first_track| {
+        let [x, y] = first_track.resolution.unwrap();
+        if x >= y {
+            [1920, 1080] // Landscape
+        } else {
+            [1080, 1920] // Portrait
+        }
+    });
```

Comment:
I suppose this could be simplified by you just setting `target_resolution = 1920 * 1080`, i.e. a pixel count.